modules/clickhouse/src/main/java/org/testcontainers/containers/ClickHouseContainer.java

```diff
@@ -1,92 +1,45 @@
-    public static final String NAME = "clickhouse";
-
-    private static final DockerImageName DEFAULT_IMAGE_NAME = DockerImageName.parse("yandex/clickhouse-server");
-
-    @Deprecated
-    public static final String IMAGE = DEFAULT_IMAGE_NAME.getUnversionedPart();
-
-    @Deprecated
-    public static final String DEFAULT_TAG = "18.10.3";
-
-    public static final Integer HTTP_PORT = 8123;
-    public static final Integer NATIVE_PORT = 9000;
+public class ClickHouseContainer<SELF extends ClickHouseContainer<SELF>> extends GenericContainer<SELF> {
 
-    private static final String DRIVER_CLASS_NAME = "ru.yandex.clickhouse.ClickHouseDriver";
-    private static final String JDBC_URL_PREFIX = "jdbc:" + NAME + "://";
-    private static final String TEST_QUERY = "SELECT 1";
-
-    private String databaseName = "default";
-    private String username = "default";
-    private String password = "";
-
-    /**
-     * @deprecated use {@link ClickHouseContainer(DockerImageName)} instead
-     */
-    @Deprecated
     public ClickHouseContainer() {
-        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
+        this(ClickHouseInit.DEFAULT_TAG);
     }
 
-    public ClickHouseContainer(String dockerImageName) {
-        this(DockerImageName.parse(dockerImageName));
+    public ClickHouseContainer(String tag) {
+        this(ClickHouseInit.DEFAULT_IMAGE_NAME.withTag(tag));
     }
 
-    public ClickHouseContainer(final DockerImageName dockerImageName) {
+    public ClickHouseContainer(DockerImageName dockerImageName) {
         super(dockerImageName);
+        dockerImageName.assertCompatibleWith(ClickHouseInit.DEFAULT_IMAGE_NAME);
 
-        dockerImageName.assertCompatibleWith(DEFAULT_IMAGE_NAME);
-
-        withExposedPorts(HTTP_PORT, NATIVE_PORT);
-        waitingFor(
-            new HttpWaitStrategy()
-                .forStatusCode(200)
-                .forResponsePredicate(responseBody -> "Ok.".equals(responseBody))
-                .withStartupTimeout(Duration.ofMinutes(1))
-        );
+        ClickHouseInit.Init(this);
     }
 
     @Override
     public Set<Integer> getLivenessCheckPortNumbers() {
-        return Sets.newHashSet(HTTP_PORT);
+        return Sets.newHashSet(ClickHouseInit.HTTP_PORT);
     }
 
-    @Override
-    public String getDriverClassName() {
-        return DRIVER_CLASS_NAME;
+    public String getDatabaseName() {
+        return ClickHouseInit.databaseName;
     }
 
-    @Override
-    public String getJdbcUrl() {
-        return JDBC_URL_PREFIX + getHost() + ":" + getMappedPort(HTTP_PORT) + "/" + databaseName;
-    }
-
-    @Override
     public String getUsername() {
-        return username;
+        return ClickHouseInit.username;
     }
 
-    @Override
     public String getPassword() {
-        return password;
+        return ClickHouseInit.password;
     }
 
-    @Override
     public String getTestQueryString() {
-        return TEST_QUERY;
+        return ClickHouseInit.TEST_QUERY;
     }
```

Comment:
This method is not necessary anymore if the class does not extend `JdbcContainer` anymore. Maybe we deprecate it in this PR or remove it?

Reply:
Removed it